docs/features/async-main.md

```diff
@@ -0,0 +1,12 @@
+# Async Task Main
+## [dotnet/csharplang proposal](https://github.com/dotnet/csharplang/blob/master/proposals/async-main.md)
+
+## Technical Details
+
+* The compiler must recognize `Task` and `Task<int>` as valid entrypoint return types in addition to `void` and `int`.
```

Comment:
@sharwell What Vlad said. Since GetResult on tasklikes isn't required to block, letting async main use them is something that we'll have to think more about.